### PR TITLE
More -Wextra-semi fixes

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: Modules/IO/MeshBase/include/itkMeshConvertPixelTraits.h [hunk 1]
```diff
@@ -170,15 +170,15 @@ ITK_DEFAULTCONVERTTRAITS_NATIVE_SPECIAL(double);
...
+  }
 
 
 // Define traits for Offset<> from dimensions 1 to 5
-ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(1)
-ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(2)
-ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(3)
-ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(4)
-ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(5)
+ITK_MESH_DEFAULTCONVERTTRAITS_OFFSET_TYPE(1);
```
Comment (dzenanz) [inline Modules/IO/MeshBase/include/itkMeshConvertPixelTraits.h:177]: Here, semicolons are added. Is this intentional?

**Review by dzenanz — APPROVED** [review]

File: Modules/IO/MeshBase/include/itkMeshConvertPixelTraits.h (hunk 1 above)
Comment (seanm) [reply Modules/IO/MeshBase/include/itkMeshConvertPixelTraits.h:177]: Yes. My understanding is that ITK prefers that even non-function-like macros should require trailing semis (by means of that static_assert hack).  Not a fan of that style personally, but going with it.

**Review by thewtex — APPROVED** [review]
@seanm thanks :1st_place_medal:

**Comment by seanm** [comment]
BTW I unsuppressed -Wextra-semi on one of my cdash submissions yesterday, and no warnings today!  So we are now -Wextra-semi-clean!

**Pull request merged**